playground/server/middleware/drupalFormHandler.ts

```
@@ -0,0 +1,39 @@
+export default defineEventHandler(async (event) => {
+  const runtimeConfig = useRuntimeConfig()
+  const { ceApiEndpoint, drupalBaseUrl, serverDrupalBaseUrl } = runtimeConfig.public.drupalCe
+  const fetchUrl = serverDrupalBaseUrl ? serverDrupalBaseUrl : drupalBaseUrl
+
+  if (event.node.req.method === 'POST') {
+    const formData = await readFormData(event)
+
+    if (formData) {
+      const targetUrl = event.node.req.url
+      const response = await $fetch.raw(fetchUrl + ceApiEndpoint + targetUrl, {
+        method: 'POST',
+        body: formData,
+      }).catch((error) => {
+        event.context.drupalCeCustomPageResponse = {
+          error: {
+            data: error,
+            statusCode: error.statusCode || 400,
+            message: error.message || 'Error when POSTing form data (drupalFormHandler).',
+          }
+        }
+      })
+
+      if (response) {
+        event.context.drupalCeCustomPageResponse = {
+          _data: response._data,
+          headers: Object.fromEntries(response.headers.entries()),
+        }
+        return
+      }
```

what about the else case here. seems to be an error case which is silently ignored. we should error out properly also

---

it isn't an error but it's unnecessary so i'll remove it